### PR TITLE
Stage 2: A bunch of cleaning up

### PR DESCRIPTION
**Pull request by SpexGuy**
This fixes `std.target.Target.Cpu.Feature.feature_set_fns`, which may unlock other tests.

### CODE REVIEW

File: src/Sema.zig
Comment (SpexGuy) [inline src/Sema.zig:11752]: I don't understand what this is trying to check.  I had to change it to keep the tests passing, but I'm not sure if it's still correct.  @andrewrk can you please explain what this is for?

**Comment by SpexGuy** [comment]
Ok this technique doesn't work.  The "correct" alternative would be a pretty big change, but I think it would make a lot of things cleaner.  That change would be to make the field initializer block into its own decl, which is a child of the "namespace" decl.  From there we could start to look at removing namespace decls entirely, which I think is a good step.

There's also a hacky short-term solution I could do if anyone is immediately blocking on this.

**Comment by SpexGuy** [comment]
Alright, I ended up with a compromise.  It didn't totally make sense to make fields a decl, since they have no tv.  Instead, they now use the namespace decl, but they are analyzed in a separate sema context.  This allows us to analyze them at any time, so we don't need the array of types to process at end of scope.  I then added a namespace field to blocks, so that a block can have a narrower namespace than its containing decl.  This allows fields to be analyzed in the context of the namespace decl without making namespace decls a special case, and without creating a stub decl for the fields.

File: src/Compilation.zig
Comment (SpexGuy) [inline src/Compilation.zig:1892]: Since I was updating all of the references to decl.namespace (see the comment on that declaration for why), I also figured I'd update the ones going straight to the file scope (which is most of them) to use `getFileScope()` instead.  This will allow us to more easily change how the file scope is accessed if e.g. we move it into the decl directly or we move it to only the root namespace or something.  It also makes the codebase more searchable in general.

File: src/Module.zig
Comment (SpexGuy) [inline src/Module.zig:353]: This name was changed because the semantics have changed -- most uses should now use `block.namespace` instead.  I want this to cause compile errors on other PRs if they are using the old semantics.

File: src/Module.zig
Comment (SpexGuy) [inline src/Module.zig:4272]: The name is available at all call sites, and having it here is nice for debugging, so I adjusted it to be passed in.

File: src/Module.zig
Comment (SpexGuy) [inline src/Module.zig:625]: I was getting confused by type names like `mem.Allocator.mem.Allocator`, so this alternate method renders them as `mem/Allocator.zig:mem.Allocator`. Which now that I think about it might indicate another namespace bug... but the format conveys more information regardless. This is what the debug report uses now.

File: src/Sema.zig
Comment (SpexGuy) [inline src/Sema.zig:11838]: I considered moving `semaStructFields` and `semaUnionFields` into `Module.zig`, but they call a lot of functions that are currently private to `sema` and there isn't a clean place to split this function because it retains a lot of locals.

**Comment by SpexGuy** [comment]
With the removal of `Module.Scope`, this PR probably conflicts with literally all of the other ones.  @andrewrk once this is approved, let's coordinate on the merge.  You pull in any other PRs first, and then I'll rebase this and fix the conflicts so this can be pulled quickly as well.

File: src/Sema.zig
Comment (SpexGuy) [reply src/Sema.zig]: This was trying to check that the sema was the correct one for resolving the struct fields.  Since we now use captures for values that cross this boundary, the whole struct resolution stack is no longer needed.  It's been deleted in this PR.

**Review by andrewrk — APPROVED** [review]

**Pull request merged**
